libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/NavigationCustomEvent.kt

```diff
@@ -0,0 +1,51 @@
+package com.mapbox.navigation.core.telemetry.events
+
+import android.annotation.SuppressLint
+import android.os.Build
+import android.os.Parcel
+import com.google.gson.Gson
+import com.mapbox.android.telemetry.Event
+import com.mapbox.android.telemetry.TelemetryUtils
+import com.mapbox.navigation.base.metrics.MetricEvent
+import com.mapbox.navigation.base.metrics.NavigationMetrics
+
+@SuppressLint("ParcelCreator")
+internal class NavigationCustomEvent : Event(), MetricEvent {
+
+    private companion object {
+        private val OPERATING_SYSTEM = "Android - ${Build.VERSION.RELEASE}"
+    }
+
+    var type: String = ""
+    var payload: String? = null
+    val version: String = "2.4"
+    var customEventVersion: String = "1.0.0"
+    val event: String = NavigationMetrics.CUSTOM_EVENT
+
+    val created: String = TelemetryUtils.obtainCurrentDate()
+    var createdMonotime: Int = 0
+    val operatingSystem: String = OPERATING_SYSTEM
+    val device: String? = Build.MODEL
+    var driverMode: String = ""
+    val driverModeId: String = TelemetryUtils.obtainUniversalUniqueIdentifier()
+    val driverModeStartTimestamp: String = TelemetryUtils.obtainCurrentDate()
```

Comment:
let's set non-valid fields to be explicit. Thes fields should be set right, otherwise we can try to rely on false data
```suggestion
    val driverModeId: String = "non_valid"
    val driverModeStartTimestamp: String = "non_valid"
```

Reply:
both of these values cannot be `non_valid` because they need to match a specific pattern on the server side. Sending these values, will result in error codes. cc @RingerJK